Search/ReIndex/ResumeManager.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/*
+ * This file is part of the MassiveSearchBundle
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Massive\Bundle\SearchBundle\Search\ReIndex;
+
+/**
+ * Allow the storage of "checkpoints" which indicate how far
+ * a reindexing process has progressed in the case that it is
+ * interupted (f.e. by an error) with the aim of allowing the
+ * reindexing process to resume from where it left off.
+ *
+ * This is basically a key/value store.
+ */
+class ResumeManager
+{
+    /**
+     * Store a checkpoint. This would typically be an offset
+     * from where the reindexer can subsequently resume its
+     * task.
+     *
+     * @param mixed $name
+     * @param mixed $value
+     */
+    public function setCheckpoint($name, $value)
```

Comment:
We could inject it into the ORM listener or we could refactor
everything, which would be great. But I do not want to spend the next
week on this.

On Fri, Feb 12, 2016 at 07:56:44AM -0800, Daniel Rotter wrote:
>    In [1]Search/ReIndex/ResumeManager.php:
> 
>  > + * interupted (f.e. by an error) with the aim of allowing the
>  > + * reindexing process to resume from where it left off.
>  > + *
>  > + * This is basically a key/value store.
>  > + */
>  > +class ResumeManager
>  > +{
>  > +    /**
>  > +     * Store a checkpoint. This would typically be an offset
>  > +     * from where the reindexer can subsequently resume its
>  > +     * task.
>  > +     *
>  > +     * @param mixed $name
>  > +     * @param mixed $value
>  > +     */
>  > +    public function setCheckpoint($name, $value)
> 
>    But that's not a good thing... I.e. we have projects where loads of assets
>    are uploaded, so this functionality would be even more important for the
>    assets in these projects. Would it be hard to implement the resume
>    functionality directly in here? Would also make more sense IMO.
> 
>    —
>    Reply to this email directly or [2]view it on GitHub.
> 
>    Reverse link: [3]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/massiveart/MassiveSearchBundle/pull/74#discussion_r52758347
>    2. https://github.com/massiveart/MassiveSearchBundle/pull/74/files#r52758347
>    3. https://github.com/massiveart/MassiveSearchBundle/pull/74/files#r52758347